Plugins/PluginsShared/PluginError.swift

```diff
@@ -0,0 +1,125 @@
+import PackagePlugin
+import Foundation
+
+enum PluginError: Swift.Error, CustomStringConvertible, LocalizedError {
+    case incompatibleTarget(targetName: String)
+    case noTargetOrDependenciesWithExpectedFiles(targetName: String, dependencyNames: [String])
+    case badArguments([String])
+    case noTargetsMatchingTargetName(targetName: String)
+    case tooManyTargetsMatchingTargetName(targetName: String, matchingTargetNames: [String])
+    case fileErrors([FileError])
+
+    var description: String {
+        switch self {
+        case .incompatibleTarget(let targetName):
+            return "Incompatible target called '\(targetName)'. Only Swift source targets can be used with the Swift OpenAPI Generator plugin."
+        case .noTargetOrDependenciesWithExpectedFiles(let targetName, let dependencyNames):
+            let introduction = dependencyNames.isEmpty ? "Target called '\(targetName)' doesn't contain" : "Target called '\(targetName)' or its local dependencies \(dependencyNames) don't contain"
```

Comment:
```suggestion
            let introduction = dependencyNames.isEmpty ? "Target called '\(targetName)' doesn't contain" : "Target called '\(targetName)' or its local dependencies \(dependencyNames.joined(separator: ", ") don't contain"
```

Should we format the names somehow?

Reply:
I did want to format dependencies, but preferred to skip it since it would have been a tiny bit complicated.
I'll see what i can do, probably just the same thing as `.joined(separator: ", ")` but with a "or" as the last separator to make it sound more human.